### PR TITLE
Outlook set signature using event-based activation 

### PR DESCRIPTION
**Pull request by davidchesnut**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                               |
| New feature?    | yes                              |
| New sample?     | yes                               |
| Related issues? | no |

## What's in this Pull Request?

New sample to show how to use event-based activation in Outlook. Scenario shows how add-in can prompt user even when task pane is not showing. And lets user enter and save a signature that gets applied to messages and appointments.

### CODE REVIEW

File: Samples/outlook-set-signature/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  <LaunchEvent Type="OnNewMessageCompose" FunctionName="checkSignature" />
+  <LaunchEvent Type="OnNewAppointmentOrganizer" FunctionName="checkSignature" />
+</LaunchEvents>
+```
+
+### Handling the events
+
+When the user creates a new message or appointment, Outlook will load the files specified in the manifest to handle the `OnNewMessageCompose` and `OnNewAppointmentOrganizer` events. Outlook on the web will load the `autorunweb.html` page, which then also loads `autorunweb.js` and `autorunshared.js`.
+
+The `autorunweb.js` file contains a version of the `insert_auto_signature` function used specifically when running on Outlook on the web. The [setSignatureAsync() API cannot be used in Outlook on the web for appointments](https://docs.microsoft.com/javascript/api/outlook/office.body?view=outlook-js-preview#setSignatureAsync_data__options__callback_). Therefore `insert_auto_signature` inserts the signature into a new appointment by directly writing to the body text of the appointment.
```
Comment (lindalu-MSFT) [inline Samples/outlook-set-signature/README.md:118]:
```suggestion
The `autorunweb.js` file contains a version of the `insert_auto_signature` function used specifically when running on Outlook on the web. The [setSignatureAsync() API cannot be used in Outlook on the web for appointments](https://docs.microsoft.com/javascript/api/outlook/office.body?view=outlook-js-preview#setSignatureAsync_data__options__callback_). Therefore, `insert_auto_signature` inserts the signature into a new appointment by directly writing to the body text of the appointment.
```

File: Samples/outlook-set-signature/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+The `autorunshared.js` file contains the `checkSignature` function that handles the events from Outlook. It also contains additional code that is shared and loaded when the add-in is used in Outlook on the web and Outlook on Windows. On Outlook on Windows, this file is loaded directly and `autorunweb.html` and `autorunweb.js` are not loaded.
+
+The `autorunshared.js` file contains a version of the `insert_auto_signature` function that uses the `setSignatureAsync()` API to set the signature for both messages and appointments.
+
+Note that you can use a similar pattern when handling events. If you need code that only applies to Outlook on the web, you can load it in a separate file like `autorunweb.js`. And for code that applies to both Outlook on the web and Outlook on Windows, you can load it in a shared file like `autorunshared.js`.
```
Comment (lindalu-MSFT) [inline Samples/outlook-set-signature/README.md:124]:
```suggestion
> [!NOTE]
> You can use a similar pattern when handling events. If you need code that only applies to Outlook on the web, you can load it in a separate file like `autorunweb.js`. And for code that applies to both Outlook on the web and Outlook on Windows, you can load it in a shared file like `autorunshared.js`.
```

File: Samples/outlook-set-signature/README.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- `editsignature.html` is loaded when the task pane first opens. It lets the user enter details such as name and title for their signature.
+- `assignsignature.html` is loaded when the user saves their details from the `editsignature.html` page. It lets the user assign the signature to actions such as "new email", "reply", and "forward.
+
+## Security notes
+
+In the webpack.config.js file, a header is set to `"Access-Control-Allow-Origin": "*"`. This is only for development purposes. In production code, you should list the allowed domains and not leave this header open to all domains.
+
+You'll be prompted to install certificates for trusted access to https://localhost. The certificates are intended only for running and studying this code sample. Do not reuse them in your own code solutions or in production environments.
+
+You can install or uninstall the certificates by running the following commands in the project folder.
```
Comment (lindalu-MSFT) [inline Samples/outlook-set-signature/README.md:139]:
```suggestion
Install or uninstall the certificates by running the following commands in the project folder.
```

**Review by lindalu-MSFT — CHANGES_REQUESTED** [review]
In a few code blocks, you're missing the closing ``` which extends the code formatting to non-code text. I also moved the **Applies to:** section to the top, and added NOTE format.

File: Samples/outlook-set-signature/README.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
+---
+page_type: sample
+products:
+- office-outlook
+- office-365
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply Samples/outlook-set-signature/README.md:5]: yep, thanks!

File: Samples/outlook-set-signature/README.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+Version  | Date | Comments
+---------| -----| --------
+1.0 | 4-01-2021 | Initial release
+
+## Disclaimer
+
+**THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.**
+
+----------
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply Samples/outlook-set-signature/README.md:58]: It's an old format from legal guidelines that I'm looking into to see if we can change soon.

File: Samples/outlook-set-signature/README.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+For documentation related to this sample, see [Configure your Outlook add-in for event-based activation (preview)](https://docs.microsoft.com/office/dev/add-ins/outlook/autolaunch)
+
+> **Note:** The features used in this sample are currently in preview and subject to change. They are not currently supported for use in production environments. To try the preview features, you'll need to [join Office Insider](https://insider.office.com/join). A good way to try out preview features is to sign up for a Microsoft 365 subscription. If you don't already have a Microsoft 365 subscription, get one by joining the [Microsoft 365 Developer Program](https://developer.microsoft.com/office/dev-program).
+
+## Features
+
+- Use event-based activation to respond to events when the task pane is not open.
+- Let the user choose and set a signature for messages and appointments.
+
+## Applies to
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply Samples/outlook-set-signature/README.md:33]: fixed

File: Samples/outlook-set-signature/README.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+# (Preview) Use Outlook event-based activation to set the signature
+
+## Summary
+
+This sample uses event-based activation to run an Outlook add-in when the user creates a new message or appointment. The add-in can respond to events, even when the task pane is not open. If no signature is set, the add-in prompts the user to set a signature, and can then open the task pane for the user.
+
+For documentation related to this sample, see [Configure your Outlook add-in for event-based activation (preview)](https://docs.microsoft.com/office/dev/add-ins/outlook/autolaunch)
+
+> **Note:** The features used in this sample are currently in preview and subject to change. They are not currently supported for use in production environments. To try the preview features, you'll need to [join Office Insider](https://insider.office.com/join). A good way to try out preview features is to sign up for a Microsoft 365 subscription. If you don't already have a Microsoft 365 subscription, get one by joining the [Microsoft 365 Developer Program](https://developer.microsoft.com/office/dev-program).
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply Samples/outlook-set-signature/README.md:26]: not in docs.

**Review by lindalu-MSFT — APPROVED** [review]

**Pull request merged**